development/transitions.md

```diff
@@ -42,16 +43,14 @@ Transitions obey the following rules:
 
 Callback arguments:
 
-* callback(err, true)
-
-  The document is saved and `ok` property value is false if `err` is truthy.
-
-* callback(err)
-
-  The document is saved and `ok` property value is false if `err` is truthy.
-  Use this when a transition fails and should be re-run.
+* callback(err, needsSaving)
+  
+   `needsSaving` is true if the `change.doc` needs to be saved to db by the transition runner. For instance the transition has edited the `change.doc` in memory.    
+   `err` if truthy, the error will be added to the `changes.doc` in memory. (Note that if `needsSaving` is falsy, the doc will not be saved, so that error will not be persisted).
 
-* callback()
+Regardless whether the doc is saved or not, the transitions will all be run (unless one crashes!). 
 
-  Nothing to be done, the document is not saved and next transition continues.
-  Transitions will run again on next change.
+Ways to error : 
+- crash sentinel. When sentinel restarts, since that `change` did not record a successful processing, it will be reprocessed. Transitions that did not save anything to the `change.doc` will be rerun.
+- save an error to `change.doc` (`callback(someTruthyError, true)`): the transition exits cleanly with error, the error is saved to `change.doc`. The transition will not be rerun on this doc.
+- log an error but don't save it to doc (`callback(truthyOrFalsy, false)`): there is no record on the `change.doc` that this transition ran. That particular `change` will not go through transitions again, but if the same doc has another change in the future, since there is no record of the erroring transition having run, it will be rerun.
```

Comment:
Not sure I understand this section, because I don't know what you mean by "error". "log an error" is quite ambiguous as well, do you mean `console.log`? If so how does the callback relate, and why can the error you pass be either truthy or falsey?